### DISCUSSION TITLE
Proxied download restriction

### DISCUSSION THREAD

**Discussion by myl7** [spencerwooo/onedrive-vercel-index#222] [Ideas]
Currently in proxied download, `url` param has no limitation. Someone may access urls like <https://drive.spencerwoo.com/api/proxy?url=https%3A%2F%2Fgoogle.github.io%2Fstyleguide%2Fcppguide.html> to use the api as a proxy, which does not seem to be expected.
To ensure `url` is valid, maybe we can check if it is an index api `raw=true` redirection.

And for people who do not use it, as proxied download will consume Vercel bandwidth limitation, maybe an option to turn it off is required.

**Comment by spencerwooo** [comment]
proxied download now tries to match the `directLink` defined in `api.json` before downloading (as i believe sharepoint's direct link urls are customised)
